packages/test-base/src/jvmTest/kotlin/io/realm/kotlin/test/compiler/Utils.kt

```diff
@@ -2,7 +2,64 @@ package io.realm.kotlin.test.compiler
 
 import com.tschuchort.compiletesting.KotlinCompilation
 import com.tschuchort.compiletesting.SourceFile
+import io.realm.kotlin.compiler.CollectionType
 import io.realm.kotlin.test.util.Compiler
 
 fun createFileAndCompile(fileName: String, code: String): KotlinCompilation.Result =
     Compiler.compileFromSource(SourceFile.kotlin(fileName, code))
+
+/**
+ * Generates a formatted string containing code that can be built by the compiler.
+ * Use it to generate different types of scenarios for nullability of contained type and the field
+ * itself.
+ */
+fun getCode(
+    collectionType: CollectionType,
+    contentType: String = "",
+    nullableContent: Boolean = false,
+    nullableField: Boolean = false,
+    userStarProjection: Boolean = false
```

Comment:
Nitpick: Since this a type of "contentType", it would be nice with an API that capture that constraint.

Maybe have two variants of `getCode`:

- getCode(collectionType, contentType, nullableContent)
- getCode(collectionType, nullableContent) // or maybe getCodeWithStarProjection

or make star the default value?

```
fun getCode(
    collectionType: CollectionType,
    contentType: String = "*",
    nullableContent: Boolean = false,
    nullableField: Boolean = false
)
```

Reply:
Good point. I think I can keep everything within the same function if I make `*` the default value for `contentType`.